### DISCUSSION TITLE
[Question] What is the difference between useRouter and useNavigation?

### DISCUSSION THREAD

**Discussion by devoren** [expo/router#779] [Q&A]
### Which package manager are you using? (Yarn is recommended)

yarn

### Summary

What is the diffrence between `useRouter` and `useNavigation`? When to use `useRouter` and `useNavigation`? Can we use `useNavigation` instead of `useRouter` on web? Does `useNavigation` work the same as `useNavigation` from `react-navigation`? We also have `useRootNavigation`. Can we use `useRootNavigation` instead `useRouter` when we have this problem: 
`The action 'NAVIGATE' with payload was not handled by any navigator`

### Minimal reproducible example

`const navigation = useNavigation();`
`const router= useRouter();`

**Comment by curiosbasant** [comment]
Somebody mindful please answer.

**Comment by wwingyou** [comment]
I think, as [this link](https://docs.expo.dev/router/reference/hooks/#usenavigation) says, `useNavigation` is for calling layout specific apis.
